lib/logflare/partners.ex

```diff
@@ -0,0 +1,113 @@
+defmodule Logflare.Partners do
+  @moduledoc """
+  Handles Partner context
+  """
+  import Ecto.Query
+
+  alias Logflare.Partners.Partner
+  alias Logflare.Repo
+  alias Logflare.Users
+
+  @spec get(integer()) :: Partner.t() | nil
+  @doc """
+  Fetch single partner by given id
+  """
+  def get(id), do: Repo.get(Partner, id)
+
+  @spec new_partner(binary()) :: {:ok, Partner.t()} | {:error, any()}
+  @doc """
+  Creates a new partner with given name and token to be encrypted
+  """
+  def new_partner(name) do
+    %Partner{}
+    |> Partner.changeset(%{name: name})
+    |> Repo.insert()
+  end
+
+  @spec list_partners() :: [Partner.t()]
+  @doc """
+  Lists all partners
+  """
+  def list_partners(), do: Repo.all(Partner)
+
+  @spec list_users(Partner.t()) :: [User.t()]
+  @doc """
+  Lists all users created by a partner
+  """
+  def list_users(%Partner{token: token}) do
+    query =
+      from(p in Partner,
+        join: u in assoc(p, :users),
+        where: p.token == ^token,
```

Comment:
would be more efficient to query from user table instead and filter from there.  
as above, would be better to filter on id instead.